### ISSUE TITLE
Android Variant: Keeps Photo in Albums/Pictures

### ISSUE THREAD

**Issue by bendspoons** [NeutrinosPlatform/cordova-plugin-document-scanner#63]
On iOS, the scanner does not put the scanned image in the Pictures Folder of the iPhone, so it is only available in my App.

On Android, i think because of the "first Photo, then scan" Method, the Photo taken is placed in my Albums/Pictures Folder on Android. This is, at least in my case/app, an absolute no-go.

The Photo cannot be deleted by my App, because i only get the "scanned image" path, and not the Photo Path.

Is there a workaround like in the "Camera Plugin", to not put the image in my Albums?!

Thanks a lot!

**Comment by ChrisTomAlx** [comment]
Hey @bendspoons 

Thanks for raising this issue.

There is a folder called scanSample being created in the root directory within which the camera clicked images are stored. As of now there is no option to delete this image after scan, but as a workaround you could find this folder programmatically and then delete it using another plugin.

As for the future plan, I will add this to our enhancements backlog and provide an option for users to view both ios and android in gallery if they require it or not if they don't.

Cheers,
[Chris](https://github.com/ChrisTomAlx)
[Neutrinos](http://www.neutrinos.co/)

**Comment by bendspoons** [comment]
Hi, thanks for your reply, i can delete the scanSample Folder, and all images in it. BUT, the Picture is somehow ALSO stored/put into my /Pictures Folder, where my "regular Fotos" are stored. And the filename in this folder ist completetly different from what the app returns and the name in the scanSmaple folder. Any ideas?!

**Comment by ChrisTomAlx** [comment]
The image in the Pictures folder is the scanned image, which is the image URI that is being returned.

Cheers,
[Chris](https://github.com/ChrisTomAlx)
[Neutrinos](http://www.neutrinos.co/)

**Comment by bendspoons** [comment]
HI, sorry, i kinda forgot about this issue, because it was not that urgent, and i had other projects to manage. I'll look into this issue again

**Comment by levi730** [comment]
@bendspoons did you ever figure anything out about this?  I'm in the same situation. I know the real fix is backlogged, but I'm wondering if anybody has alternate suggestions.

Thanks!

**Comment by bendspoons** [comment]
@leci730 Nah, unfortunately nothing with which im satisfied. I remove the taken image via Code, but unfortunately the image is not removed immmediately, its kinda blurred in its folder.

**Comment by fairojm** [comment]
Hi anyone got the solution for this. i return the image as base64 still it stores the image in **Camera** and scanned image in **scanSample** and **pictures** Folder

**Comment by bastifix** [comment]
@fairojm I'm also still searching for a way to solve this.

**Comment by fairojm** [comment]
@bastifix i fixed this issue by deleting the folder and image manually. i didn't know it is the right way. i done this as a temporary fix.

**Comment by ChrisTomAlx** [comment]
@fairojm you mean via code?
Could you share the workaround code with us because I don't know when I will be able to look at this issue. In the meantime it could help someone who runs into this.

Cheers,
[Chris](https://github.com/ChrisTomAlx)
[Neutrinos](http://www.neutrinos.co/)

**Comment by fairojm** [comment]
@ChrisTomAlx i delete the image using modified time (which is latest) after the scanned image is returned from the plugin. I think its not the safe way to delete. but i don't know the another way to fix this.

**Comment by ChrisTomAlx** [comment]
I am not sure when I will be able to take a look at this issue. If anyone is up for raising a pr, I could review any pr's that come my way.

Cheers,
[Chris](https://github.com/ChrisTomAlx)
[Neutrinos](http://www.neutrinos.co/)
